Detectors/FIT/T0/simulation/src/Detector.cxx

```diff
@@ -217,90 +218,172 @@ void Detector::ConstructOpGeometry()
-  //   z=-pinstart[2] + 2*ptopref[2] + preg[2];
+  TVirtualMC::GetMC()->Gsvolu("0MTO", "BOX", getMediumID(kOpGlass), pmcptopglass, 3); //Op  Glass
+  TGeoVolume* mcptop = gGeoManager->GetVolume("0MTO");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + pmcptopglass[2];
+  ins->AddNode(mcptop, 1, new TGeoTranslation(0, 0, z));
+
+  TVirtualMC::GetMC()->Gsvolu("0MCP", "BOX", getMediumID(kAir), pmcp, 3); //glass
+  TGeoVolume* mcp = gGeoManager->GetVolume("0MCP");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + 2 * pmcptopglass[2] + 2 * preg[2] + pmcp[2];
   ins->AddNode(mcp, 1, new TGeoTranslation(0, 0, z));
+  TVirtualMC::GetMC()->Gsvolu("0MIN", "BOX", getMediumID(kGlass), pmcpinner, 3); //glass
+  TGeoVolume* mcpinner = gGeoManager->GetVolume("0MIN");
+  mcp->AddNode(mcpinner, 1, new TGeoTranslation(0, 0, 0));
+
+  TVirtualMC::GetMC()->Gsvolu("0MSI", "BOX", getMediumID(kGlass), pmcpside, 3); //glass
+  TGeoVolume* mcpside = gGeoManager->GetVolume("0MSI");
+  x = -pmcp[0] + pmcpside[0];
+  y = -pmcp[1] + pmcpside[1];
+  mcp->AddNode(mcpside, 1, new TGeoTranslation(x, y, 0));
+  x = pmcp[0] - pmcpside[0];
+  y = pmcp[1] - pmcpside[1];
+  mcp->AddNode(mcpside, 2, new TGeoTranslation(x, y, 0));
+  x = -pmcp[1] + pmcpside[1];
+  y = -pmcp[0] + pmcpside[0];
+  mcp->AddNode(mcpside, 3, new TGeoCombiTrans(x, y, 0, new TGeoRotation("R2", 90, 0, 0)));
+  x = pmcp[1] - pmcpside[1];
+  y = pmcp[0] - pmcpside[0];
+  mcp->AddNode(mcpside, 4, new TGeoCombiTrans(x, y, 0, new TGeoRotation("R2", 90, 0, 0)));
+
+  TVirtualMC::GetMC()->Gsvolu("0MBA", "BOX", getMediumID(kCeramic), pmcpbase, 3); //glass
+  TGeoVolume* mcpbase = gGeoManager->GetVolume("0MBA");
+  z = -pinstart[2] + 2 * pal[2] + 2 * ptopref[2] + pmcptopglass[2] + 2 * pmcp[2] + pmcpbase[2];
+  ins->AddNode(mcpbase, 1, new TGeoTranslation(0, 0, z));
+
+  // Al Housing for Support Structure
+  //  ins->AddNode(alsup,1);
 }
 
 Bool_t Detector::ProcessHits(FairVolume* v)
 {
+  TVirtualMCStack* stack = fMC->GetStack();
   Int_t quadrant, mcp;
   if (fMC->IsTrackEntering()) {
     float x, y, z;
     fMC->TrackPosition(x, y, z);
     fMC->CurrentVolID(quadrant);
     fMC->CurrentVolOffID(1, mcp);
     float time = fMC->TrackTime() * 1.0e9; //time from seconds to ns
-    int trackID = fMC->GetStack()->GetCurrentTrackNumber();
+    int trackID = stack->GetCurrentTrackNumber();
     int detID = 4 * mcp + quadrant - 1;
     float etot = fMC->Etot();
     int iPart = fMC->TrackPid();
     float enDep = fMC->Edep();
+    Int_t parentID = stack->GetCurrentTrack()->GetMother(0);
     if (fMC->TrackCharge()) { //charge particles for MCtrue
       AddHit(x, y, z, time, 10, trackID, detID);
     }
     if (iPart == 50000050) // If particles is photon then ...
     {
       if (RegisterPhotoE(etot)) {
-        AddHit(x, y, z, time, enDep, trackID, detID);
+        //        AddHit(x, y, z, time, enDep, trackID, detID);
+        AddHit(x, y, z, time, enDep, parentID, detID);
```

Comment:
concerning the parentID, I am wondering if you are -- in fact -- only caring about the first parentID of the track "entering" a sensitive volume? (Now I think the parent could be another particle created in T0).

A somewhat related question is if you need to add a new detailed hit for each energy deposit, or if you could simply accumulate hits for the same parentID together? In this way, one could save some memory as currently there seem to be many hits created.

Reply:
Dear Sandro,
as a hit I am collecting Cherenkov photons (taking into account quantum efficiency of photocathode) and each one with its time should be saved and used in Digitizer for calculating event time  and MB trigger . Parents are important to distinguish hits from  EM interaction and estimate bias of MB trigger. We have experience   from Run2 Pb-Pb that income of EM component is  very important  and such simulation will help to set correct thresholds.
So as MC truth I need to know how many charge particles from each source produced Cherenkov photons which we registered.

Best regards
Alla